src/components/TodoItem.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Errors } from '../types/Errors';
+
+interface Props {
+  todo: Todo,
+  currentId: number | null,
+  handleDeleteTodo: (todo: Todo) => Promise<void>;
+  handleUpdateTodo: (todo: Todo) => Promise<Todo | void>,
+  setErrorMessage: (message: Errors | '') => void,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  currentId,
+  handleDeleteTodo,
+  handleUpdateTodo,
+  setErrorMessage,
+  setTempTodo,
+  setTodos,
+}) => {
+  const { title, completed } = todo;
+  const [newTitle, setNewTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleDelete = async () => {
+    setIsDeleting(true);
+
+    try {
+      await handleDeleteTodo(todo);
+    } catch {
+      setErrorMessage(Errors.DeleteTodoError);
+    } finally {
+      setIsDeleting(false);
+    }
```

Comment:
you are already handling errors inside handleDeleteTodo. You don't need to do it twice

Reply:
please check the whole code and remove duplications

Reply:
if I remove the catch in the function handleSubmit I fail another test